docs/diagrams/CompanyDetailPanelActivityDiagram.puml

```diff
@@ -0,0 +1,27 @@
+@startuml
+skin rose
+skinparam ActivityFontSize 15
+skinparam ArrowFontSize 12
+start
+:User executes command;
+
+'Since the beta syntax does not support placing the condition outside the
+'diamond we place it as the true branch instead.
+
+if () then ([view, edit or add command])
+    :setCurrentViewedCompany(Company company);
+    :clear UniqueCompanyList
+    and add the company to
+    UniqueCompanyList;
```

Comment:
perhaps you might want to separate this out into two separate actions for better clarity. i.e clear UniqueCompanyList, then Add company to UniqueCompanyList

Reply:
done